pkg/infra/main.go

```diff
@@ -92,17 +95,9 @@ func gameLoop(globalState state.State, agentMap map[string]agent.Agent, gameConf
 			shieldLoot[i] = globalState.CurrentLevel * uint(rand.Intn(3))
 		}
 
-		for i, agentState := range globalState.AgentState {
-			allocatedWeapon := rand.Intn(len(weaponLoot))
-			allocatedShield := rand.Intn(len(shieldLoot))
+		new_global_state := api.AllocateLoot(globalState, weaponLoot, shieldLoot)
```

Comment:
I think because the loot allocation mechanism is going to change quite dramatically given that it's going to be dealt as a common pool resource, it might be worth moving it to something like AgentLootDecisions, since the random allocation was only for MVP.